src/signing/interactors/batch_signing_response.rs

```diff
@@ -0,0 +1,16 @@
+use crate::prelude::*;
+
+/// The response of a batch signing request, either a Parallel or Serial signing
+/// request, matters not, because the goal is to have signed all transactions with
+/// enough keys (derivation paths) needed for it to be valid when submitted to the
+/// Radix network.
+#[derive(Clone, PartialEq, Eq, derive_more::Debug)]
+#[debug("BatchSigningResponse {{ signatures: {:#?} }}", signatures.values().map(|f| format!("{:#?}", f)).join(", "))]
+pub struct BatchSigningResponse {
+    pub signatures: IndexMap<FactorSourceIDFromHash, IndexSet<HDSignature>>,
```

Comment:
So this is a response received from the Host for a given batch signing? Shouldn't we have signatures per tx_intent as opposed to being flattened:
```rust
IndexMap<FactorSourceIDFromHash, IndexMap<TXIntent, HDSignature>>,
```

Reply:
Yes we can do that. It is  needed thought since the `HDSignature` references the FactorSourceID used to produce it and the `intent_hash` signed. So actually it could be an `IndexSet<HDSignature>` . I will take a look and see if there is any advantage changing the format.

Reply:
TODO see if we wanna change representation

Reply:
ohh wait, I didn't check the contents of HDSignature, didn't know that it actually contains a reference to the original intent_hash, I would then maybe change its name as I thought that it was just the signature.

Reply:
`HDSignatureByFactor` ?

Reply:
`SignInputAndOutput`?